src/Libraries/CoreNodeModels/Input/CustomSelection.cs

```diff
@@ -0,0 +1,96 @@
+using System.Collections.Generic;
+using System.Linq;
+
+using Autodesk.DesignScript.Runtime;
+
+using Dynamo.Graph.Nodes;
+
+using ProtoCore.AST.AssociativeAST;
+
+namespace CoreNodeModels.Input
+{
+    /// <summary>
+    /// This node allow the user to create a dropdown menu with an with an arbitrary number of customization items
+    /// </summary>
+    [NodeName("Custom Dropdown Menu")]
+    [NodeCategory(BuiltinNodeCategories.CORE_INPUT)]
+    [NodeDescription("CustomSelectionNodeDescription", typeof(Properties.Resources))]
+    [NodeSearchTags("CustomSelectionSearchTags", typeof(Properties.Resources))]
+    [OutPortNames("value")]
+    [OutPortTypes("var")]
+    [OutPortDescriptions("Selected value")]
+    [IsDesignScriptCompatible]
+    public class CustomSelection : DSDropDownBase
+    {
+        /// <summary>
+        /// Construct a new Custom Dropdown Menu node
+        /// </summary>
+        public CustomSelection() : base("Value")
+        {
+            // TODO: This isn't done in RevitDropDown. Necessary?
+            ArgumentLacing = LacingStrategy.Disabled;
```

Comment:
I don't know enough about what this affects to know whether it should be here.

Reply:
Now I do. Looks like all components without inputs have lacing disabled, I think just to avoid an empty for loop. This should probably be set on the Revit dropdowns as well?